pkg/collector/python/inter.go

```diff
@@ -0,0 +1,42 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build python
+
+package python
+
+import "sync"
+
+const maxInternerStrings = 1000
```

Comment:
TODO: Is this a reasonable size? Maybe using the same size as the interner in the statsd server would be better?

Reply:
I think we'll need telemetry to know what a good size for this would be.

"total unique strings returned from python" isn't something we can even get a good guess for from our existing telemetry.

Reply:
Sure, we can add some telemetry. But do you think we need it right away, or could this be merged and tested in staging beforehand?

Reply:
can definitely merge first and then add telemetry and try to size it 👍 
1k seems like a reasonable size to start with, I expect it will need to be tuned up, but this should still be an improvement.